### ISSUE TITLE
100% CPU when opening a project or 'Settings > Packages'

### ISSUE THREAD

**Issue by joecritch** [facebook/nuclide#84]
Hey, I've got no other packages activated inside of Atom, and after installing `nuclide-installer` package, I'm getting a 100% CPU. This is also happening when I try to view the Packages list inside of Atom, so I can't uninstall using the Prefs screen.

A brief issue, as I'm unfamiliar with the reporting process, but please let me know what information you require.

**Comment by bolinfest** [comment]
Are you on Windows, Mac, or Linux? Anything else about your system that might be special?

**Comment by joecritch** [comment]
.@bolinfest Mac, OS X Yosemite. My atom settings are symlinked, in the style of https://github.com/holman/dotfiles. Interestingly, the CPU still caps when I run atom with `$ atom --safe`. If you can point me to where I need to go, perhaps I can do some CPU profiling?

**Comment by Sterbic** [comment]
Same issue here on OS X 10.10.3 and Atom 0.211.0.

I installed nuclide through `apm` and now Atom Helper and npm max out their thread each time I start Atom. Loading the packages view in the settings tab take a few minutes.

After the first restart I had only 6 nuclide-* packages installed, after the second restart I see 9 packages so Atom Helper might be doing useful work. I still don't see some packages like `nuclide-quick-open`.

**Comment by bolinfest** [comment]
The `nuclide-installer` package goes through and runs `apm install` for each package in https://github.com/facebooknuclideapm/nuclide-installer/blob/master/lib/config.json. We're getting some reports that this is very, very slow. I'm also checking with GitHub to see if there's anything on their end that could be contributing to this.

**Comment by bolinfest** [comment]
You could experiment with installing each package through Atom's GUI to see if the behavior is any different. I need to dig into the code in `settings-view` to see if anything extra special happens there.

**Comment by Sterbic** [comment]
I am trying to uninstall everything through Atom's GUI but it's extremely slow and unresponsive. I will try a fresh install through the GUI and from source to check if there is any difference.

**Comment by bolinfest** [comment]
If you backup your `~/.atom/packages` directory, that's the easiest way to remove all of your packages.

**Comment by Sterbic** [comment]
Even uninstalling a single package after disabling it takes several minutes and `Atom Helper` goes 100%. Shouldn't this be as fast as `rm -r pkg_dir`?

Building from source with `./scripts/dev/setup` does solve the issue.

```
START INSTALL: 2015-06-24 01:17:17.482524
...
FINISH INSTALL: 2015-06-24 01:19:34.594759
PackageManager.install() took 137 seconds.
```

Now it takes less than 10 seconds to start Atom and load all packages, including 13 nuclide-* packages.

**Comment by Sterbic** [comment]
When installed from source, uninstalling through the GUI is almost instant as expected.

This is probably an issue with Atom and not directly nuclide: only the first uninstalled package is removed from the installed packages list, the other packages stay listed until I enter a different search term although they are not present in `~/.atom/packages` any more.

**Comment by Sterbic** [comment]
I tried to install `nuclide-installer` through Atom's GUI and monitor the progress through 
`ls ~/.atom/packages/`. It took about 40 minutes to complete, but the 100% cpu issue is still there.

**Comment by mauriciord** [comment]
Same problem for me. Installing with ./scripts/dev/setup is perfect now, but i want the apm install nuclide-installer.
Did you tried sudo apm install nuclide-installer ?

**Comment by Sterbic** [comment]
I will give it a try, but I doubt it will make any difference.

**Comment by grabbou** [comment]
No difference unfortunately. CPU 100% all the time, even when browsing files.

**Comment by mcz** [comment]
Same here. "Atom Helper" is running on 100% CPU on OS X 10.10.3.

**Comment by grabbou** [comment]
Just to add few more bits, I've tried installing both from Atom and from `apm`. Neither of them completed successfully as the Atom UI was just frozen. Afraid of the high CPU temperature for a few minutes, I've decided to build from source. The installation has finished w/o any problems, and all the components were there, but the whole UI was full of lags and the CPU usage was still as high as during the installation process. Decided to remove all the packages by using `apm` as the editor was completely unusable.

**Comment by zacharytamas** [comment]
Same here. Atom is basically unusable right now. Major lag: I can finish a whole line of code before it actually appears on screen. I'm having to remove the packages one by one with apm just to restore it well enough to actually work today. OS X 10.10.3

**Comment by kaizhang** [comment]
Same here. OS X 10.10.3.

**Comment by samueloph** [comment]
I have a dirty workaround for this bug:
It looks like removing atom and all its config. files resolves the problem [of course, making a new clean installation later].

You can check the proper atom's bug report here:
https://github.com/atom/settings-view/issues/555.

**Comment by mauriciord** [comment]
I removed atom and its config several times, and i made a new clean installation later.
Didn’t work for me.

Now i will try do the same with the new Atom 1.0.

Mauricio R. Duarte.
[ mauriciord@me.com <mailto:mauriciord@me.com>  |  mauriciotem@gmail.com <mailto:mauriciotem@gmail.com> ]

**Comment by chibipirate** [comment]
Try apm uninstall * for any nuclide package you see. That seemed to do the trick for me. Keep in mind that I have yet to reinstall Nuclide.

**Comment by samueloph** [comment]
@chibipirate Could you confirm if reinstalling Nuclide didn't make atom crash again? I'm starting to think this problem is related to Nuclide, considering the number of people that are participating in the settings-view's bug report.

I reinstalled nuclide here and atom started to crash again.

**Comment by rpedroni** [comment]
Same here on OS X 10.10.3. I really do believe this is nuclide related since this has only happened after installation. For me, opening the settings panel starts the havoc

**Comment by cesarandreu** [comment]
+1

After installing Nuclide, opening my settings shoots up CPU usage.

**Comment by admmasters** [comment]
Same. Only happening with Nuclide. Unusable for 5 or minutes after starting the app.

**Comment by gothy** [comment]
AtomHelper per opened window consumes 100% CPU even after closing the "Settings" tab. Waiting for more than 5minutes now...
MacOS X Yosemite.

**Comment by spencerlyon2** [comment]
Seeing this also -- posting so I'm notified of new comments

**Comment by despairblue** [comment]
@spencerlyon2 there is a subscribe button on the right :wink: 

![screen shot 2015-07-17 at 16 36 58](https://cloud.githubusercontent.com/assets/927609/8749488/238481c6-2ca2-11e5-8e2a-66c3fb263124.png)

**Comment by bneiluj** [comment]
Any news on this issue ? 
"Atom Helper" is running on > 100% CPU on OS X 10.10.4 and Atom 1.0.15

**Comment by bneiluj** [comment]
I am sorry but for now the only answer is: `rm -rf ~/.atom/packages/nuclide-*`

**Comment by despairblue** [comment]
Have you tried the manual installation using `scripts/dev/setup`? That prevents this for me.

**Comment by mauriciord** [comment]
WOW ! Same here, i tried here on OS X El Captain, and my CPU 100% easy easy easy.
I think i have a Mac Pro to solve this problem hahaha !
Impossible to use Nuclide here.

Em 21 de set de 2015, à(s) 08:24, Julien Bouteloup <notifications@github.com> escreveu:

rm -rf ~/.atom/packages/nuclide-*

**Comment by derekvanvliet** [comment]
I'm having the same problem after doing a fresh install of Atom and Nuclide on both OS X and Win7. Unfortunately this makes it pretty unusable for me.

**Comment by samueloph** [comment]
This has been happening since June 23, sincerely I don't think theres anybody working on this, I've dropped Nuclide long ago.

Can confirm this https://github.com/facebook/nuclide/issues/84#issuecomment-141945823 work though.

**Comment by bneiluj** [comment]
Weird because @ssorallen touched the repo 6 days ago. 
Why maintaining a project if nobody can use it?

**Comment by despairblue** [comment]
You can still use the [manual installation method](https://github.com/facebook/nuclide#building-from-source). That prevents this issue. I'd also liked to see this fixed, but it seems that noone even found out what the problem is: https://github.com/atom/settings-view/issues/551#issuecomment-114804300. When loading the packages [it's reading a huge load of files](https://github.com/atom/settings-view/issues/551#issue-90517487), I'm not sure why.

**Comment by derekvanvliet** [comment]
@despairblue thanks for the work around. That worked.

So this is actually an issue in Atom? Not Nuclide?

**Comment by despairblue** [comment]
@derekvanvliet I'm not sure, I haven't found the time to debug this, but if you do, I'm sure a lot of people would be grateful :smiley:

**Comment by DonSanto** [comment]
I'm having this problem too. posting so I'm notified of new comments.

**Comment by nguyenhuy** [comment]
(@DonSanto You can subscribe to this thread using the "Subscribe" button on the right side. Sorry for spamming, everyone :) )

**Comment by DonSanto** [comment]
@nguyenhuy Thanks, I've been looking for that functionality but have not been able to find the button. Now I know where it is. :+1:   (Sorry for spamming, just wanted to thank the guy :) )

**Comment by cnjsstong2** [comment]
Same here. 

OS X 10.10.5, Atom 1.0.15, installed nuclide from GUI.

**Comment by pookah-net** [comment]
Just had to restart my MBP; totally unresponsive, dialog indicates .context is missing, not sure why? (Permissionss look okay.) Posting to get notifications!

/Ian

**Comment by GameGamer43** [comment]
I have the same issue. It appear that if I remove all the nuclide packages from atom and then install from source, the issue is resolved and the load times for Atom greatly improve.

**Comment by pookah-net** [comment]
I ended up deleting nuclide in its entirety. I didn't want to spend an afternoon troubleshooting and after awhile, it just seemed that getting rid of nuclide was the easier option! Especially as most of the features from it that I was using were available in other packages.

**Comment by bolinfest** [comment]
I was doing some monitoring, and it seems like Atom sends individual requests to atom.io for every third-party package you have installed to see if you have the latest version of the package or not. Obviously when you have Nuclide installed, you have a lot of packages, so all of these requests (and presumably DOM updates as the requests come back) seem to be what upset Atom. Presumably things would be better if Atom had a batch API for this. /cc @thedaniel @nathansobo

**Comment by adamdicarlo** [comment]
@bolinfest That sounds pretty inefficient, but I'm leaning toward it being mostly due to way too much JS code, as installing via nuclide-installer ends up adding > 2GB of packages :frowning: ... see https://github.com/facebook/nuclide/issues/203

**Comment by thedaniel** [comment]
@bolinfest You're right that there's a lot of opportunity to make settings-view more efficient, and we're down to add API endpoints to the package API as well to reduce the number of requests, etc. An issue on atom/settings-view would be a good place to start.

**Comment by ku-kueihsi** [comment]
The only workaround is manual installation, according to despairblue?

**Comment by despairblue** [comment]
@ku-kueihsi it seems so, though I haven't investigated any further, since, together with http://gh-release-watch.com/ (to remind me to update nuclide when a new release happens), the manual installation poses not that much of an additional effort. Simply cd in the repos directory and issue `git pull && git checkout v0.0.x && scripts/dev/setup` and that's it.

Though I'm still curious what the actual culprit is. I do have a lot of packages installed, but only when I install the nuclide-installer this issue appears, none of the other packages provoke this.

**Comment by yubozhao** [comment]
tried @despairblue 's manual installation...Atom failed to open now.... Going to drop Nuclide until this resolve.

Setup:  Mac 10.11, Atom 1.0.19.  Clean Atom environment without any community packages.

**Comment by despairblue** [comment]
The first start after the installation can take a couple of minutes, while
atom does whatever it does.  After that it starts fine.

On Mon, Oct 12, 2015, 01:32 Bozhao <notifications@github.com> wrote:

> tried @despairblue <https://github.com/despairblue> 's manual
> installation...Atom failed to open now.... Going to drop Nuclide until this
> resolve.
>
> Setup: Mac 10.11, Atom 1.0.19. Clean Atom environment without any
> community packages.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/nuclide/issues/84#issuecomment-147255370>.
>
-- 
--------------------------------------------
Q: Why is this email five sentences or less?
A: http://five.sentenc.es

**Comment by yubozhao** [comment]
@despairblue I wish that is the case.  I test with my previous atom setup, and a clean atom setup (no additional package).  The atom failed to start and cpu shot through the roof

**Comment by crobinson** [comment]
I deleted all my nuclide package and use the @despairblue solution. It works some times, but some times make my machine slow again. Anybody know about the run button on F8 video that open and test on iOS simulator? Im using nuclide and atom but i still using Xcode to debug and test. How can i show that F8 video toolbar? i think have Xcode closed during my coding could help....

**Comment by panlw** [comment]
100% CPU, Waiting for stable release.

**Comment by ku-kueihsi** [comment]
Manual clean install. Still freezes atom. I have to delete all the plugins.

**Comment by javierbq** [comment]
Same problem, nuclide made atom unusable.

**Comment by astannard** [comment]
Hi I had the same problem, I uninstalled the packages through apm and kept retesting the editor. It started working properly once I uninstalled nuclide-diff-view, nuclide-buck-files, nuclide-clang-atom. It is probably only one of those plugins that is causing it though.

Now the only nuclide plugins I have left are:
nuclide-code-format@0.0.32
nuclide-find-references@0.0.32
nuclide-format-js@0.0.32
nuclide-ocaml@0.0.32
 nuclide-toolbar@0.0.32
nuclide-type-hint@0.0.32

Maybe when I get change I will reinstall nuclide the uninstall the three mentioned plugins one at a time until I find the culprit

**Comment by Willyham** [comment]
Sorry for not contributing to the actual issue, but for those looking to uninstall packages and can't load the GUI, you can run `apm list | grep nuclide | sed  's/├── //' | xargs apm uninstall`

**Comment by jeromecovington** [comment]
Hm. AtomHelper currently at 105%. Would really like the benefit of using the nuclide packages here, but...

**Comment by imcoddy** [comment]
+1

OS 10.10.5. Clean Atom environment without any community packages.

**Comment by mhagmajer** [comment]
+1 OS X 10.11.1 having fresh Atom install with nuclide-installer installed makes Atom unusable. I came back to installing just nuclide-flow

**Comment by z-ax** [comment]
Hey, also +1 to this issue, latest Atom release itself runs finely, but with `nuclide-installer` v0.0.35 - huge CPU usage.

Have latest OS X, MBP.

**Comment by AndreaCatalucci** [comment]
+1, atom1.2.4 on el capitan

**Comment by nowaits** [comment]
+1, atom1.3.0beta6 on El Capitan

**Comment by shoreflyer** [comment]
Hi, having the same issue (El Capitan, Atom 1.2.4), had to manually remove all nuclide packages from my ~/.atom/packages directory.
Just as a note, emptying the recycle bin after showed over 250k files being deleted... npm goodness? Yes, my recycle bin was empty before I binned the nuclide packages...

What version are you using internally if it's not behaving this way?

**Comment by cbrwizard** [comment]
+1, editor becomes unusable on my mac after installing nuclide - 104% CPU

**Comment by jadedevin13** [comment]
+1

**Comment by adrice727** [comment]
+1

**Comment by clude** [comment]
+1

**Comment by LukeeeeBennett** [comment]
+1 atom1.2.4 on El Capitan

@bolinfest has highlighted the need to integrate a batch package install api in the official atom repo, has anyone followed this up with an issue on the atom repo? Lastly, has anyone started on a pr?

**Comment by lukeed** [comment]
Would it make sense to have nuclide just be a fork of Atom, instead of a layer on top of it?

Still allow apm to run on nuclide, just restructure whatever needs restructuring internally, including the NPM module de-duping.

Shrug -- early morning fog rants...

**Comment by plrthink** [comment]
same problem here

**Comment by mauriciord** [comment]
I stopped using Nuclide because of this, I'm just using JSX React Snippets
for Atom and other individually, not the created by Facebook.
Working for me this way.
I hope it helps, thanks.

Atenciosamente,
Maurício R. Duarte.

2015-12-07 8:43 GMT-02:00 Perry Poon <notifications@github.com>:

**Comment by LukeeeeBennett** [comment]
@lukeed - I know what you're getting at, forking Atom would mean that those interested in the nuclide behemoth of a package can focus purely on redesigning and redeveloping its infrastructure to perfectly suit nuclides needs. Personally however, I believe nuclide could benefit from the wider Atom community if it stayed as purely an abstraction package for Atom. Additionally, Atom could benefit from nuclide as the nuclide community helps Atom become more suitable and stable for larger package suites.

Maybe something that should be looked into is the fact that actually the nuclide package in whole is a bit too much and seems to target a very specific use case/user story. In my opinion, I would love to see the 'nuclide installer' as the root Atom package, from there it is an abstracted settings and package management panel purely for the nuclide packages. You could reply "Well... you can just directly download the packages you want and only them" but this is a problem of accessibility and usability and I believe that could be of benefit. **This is just my opinion**, of course there is potential for this to be a very stupid opinion.

**Comment by ssorallen** [comment]
> In my opinion, I would love to see the 'nuclide installer' as the root Atom package, from there it is an abstracted settings and package management panel purely for the nuclide packages.

@LukeeeeBennett, stay tuned. We have been exploring that direction to address the many valid critiques of the current approach that make Nuclide difficult to install and update. More updates when we've got something to show.

**Comment by ads1018** [comment]
+1

**Comment by Dids** [comment]
I can verify that manually installing Nuclide from source doesn't cause any of these issues.

**Comment by luismreis** [comment]
Manually installing from source works for me too, but now I've got either one or two tree views. :stuck_out_tongue_closed_eyes:

**Comment by lukeed** [comment]
How are you installing from source? The readme no longer contains those instructions. And after finding them in the readme's history, running `scripts/dev/setup` didn't do anything except `npm install`

**Comment by mauriciord** [comment]
# Clone the source $ git clone https://github.com/facebook/nuclide.git
$ cd nuclide
# Install dependencies $ npm install # Link the 'nuclide' package to Atom's
package directory $ apm link

Atenciosamente,
Maurício R. Duarte.

2015-12-29 21:26 GMT-02:00 Luke Edwards <notifications@github.com>:

**Comment by lukeed** [comment]
@mauriciord gotcha, thanks!

**Comment by efkan** [comment]
+1 for Windows 10

**Comment by mostafaeweda** [comment]
This should have been fixed with the UP release, http://nuclide.io/blog/2016/01/13/Nuclide-v0.111.0-The-Unified-Package/

**Comment by z-ax** [comment]
Finally! Checked, works as expected, all good.

**Comment by mauriciord** [comment]
Finally working ! Thanks

**Comment by v3ss0n** [comment]
There is 100% IO usage when settings is Open and Nuclided is installed. It spawns about 4-5 watchman processes that takes 100% I/O .  Latest nuclide.

**Comment by mostafaeweda** [comment]
@v3ss0n are you using the latest Atom? We are currently fine at Atom 1.6.2.
However, there is a known issue (pending fix) with Atom >= 1.7 that causes such CPU spike.

**Comment by v3ss0n** [comment]
i am with atom 1.7 +

**Comment by mostafaeweda** [comment]
the fix will be out in ~ 2 days

**Comment by eworkflow** [comment]
My Atom started crashing on its startup. After booting it in the safe mode (atom --safe) I attempted to uninstall all the Nuclide packages but most of them resulted in the Atom uninstalling errors. When deleting the related directories manually, I noticed that some of the "Nuclide-*" directories have obscene sizes such as ~80MB. I fear that the "hacking" packages got hacked themselves unless there is another explanation for this nonsense.
